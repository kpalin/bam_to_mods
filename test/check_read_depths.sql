create table bam_mods as
select
    *
from
    read_csv('test/tmp/A+a.0.out');

create table depths as (
    SELECT
        *,
        '1' as haplotype,
        pos as 'end'
    from
        read_csv(
            "data/fibreseq_demo_pacbio_hp1.depth.gz",
            names = ["#chromosome","pos","depth"]
        )
    UNION
    SELECT
        *,
        '2' as haplotype,
        pos as 'end'
    from
        read_csv(
            "data/fibreseq_demo_pacbio_hp2.depth.gz",
            names = ["#chromosome","pos","depth"]
        )
);
.mode csv
select
    count(*),
(called_reads + uncalled_reads + mismatch_reads - depth) as gap
from
    bam_mods a,
    depths b
where
    b."end" = a."end"
    and a.haplotype = b.haplotype
    
group by 
    gap;


