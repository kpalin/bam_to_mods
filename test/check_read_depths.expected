count_star(),gap
16,-2
311,-1
5704,0
235,1
10,2
